File: C#/src/Bruteforce.cs
using System;
using System.Linq;

namespace Brutefarce
{
    class Bruteforce
    {

        private const string Alphanumberic = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
        private string Password;
        private int MinLength;
        private int MaxLength;
        private int Guesses = 1;
        private bool IsNotCracked = true;
        private Random Random;

        public Bruteforce(string password, int minLength, int maxLength)
        {
            Password = password;
            MinLength = minLength;
            MaxLength = maxLength;
            Random = new Random();
        }

        private int GuessLength()
        {
            return Random.Next(MinLength, MaxLength);
        }

        private string NewGuess(int length)
        {
            return new string(Enumerable.Range(1, length).Select(_ => Alphanumberic[Random.Next(Alphanumberic.Length)]).ToArray());
        }

        public int BreakPassword()
        {
            while (IsNotCracked)
            {
                int length = GuessLength();
                string currentGuess = NewGuess(length);

                Console.WriteLine("{0}. Guess is {1}", Guesses, currentGuess);

                if (currentGuess.Equals(Password))
                {
                    Console.WriteLine("Found a match! {0}", currentGuess.ToString());
                    IsNotCracked = false;
                }

                Guesses++;
            }

            return Guesses;
        }
    }
}

File: Nim/src/bruteforce.nim
import random

proc generateGuessLength(minLength: int, maxLength: int) : int =
  result = rand(minLength .. maxLength)

proc generateNewGuess(guessLength: int) : string =
  for _ in 1 .. guessLength:
    add(result, char(rand(int('A') .. int('z'))))

proc bruteforce(minLength: int, maxLength: int, password: string) : int =
  var isNotCracked = true
  var guesses: int = 1

  while isNotCracked:
    let guessLength: int = generateGuessLength(minLength, maxLength)
    let guess: string = generateNewGuess(guessLength)

    echo guesses, ". Guess is ", guess

    if guess == password:
      echo "Found a match! ", guess
      isNotCracked = false
      
    guesses += 1

  result = guesses